src/mlang/backend_ir/bir_interface.ml

```diff
@@ -327,10 +327,11 @@ let adapt_program_to_function (p : Bir.program) (f : bir_function) :
         Pos.same_pos_as (Bir.SVerif cond) cond.cond_expr :: acc)
       f.func_conds []
   in
-  ( {
-      p with
-      statements =
-        unused_input_stmts @ const_input_stmts @ p.statements @ conds_stmts;
-      outputs = f.func_outputs;
-    },
+  let mpp_functions =
+    Bir.FunctionMap.add p.main_function
+      (unused_input_stmts @ const_input_stmts @ Bir.main_statements p
+     @ conds_stmts)
+      p.mpp_functions
```

Comment:
This is fine for now, but we probably don't want initialisation and verifications to be included in what will become an exposed function. Please add a comment stating we will want to change this.